src/Components/WebAssembly/WebAssembly/src/Services/WebAssemblyDynamicResourceLoader.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using System.Runtime.Loader;
+using Microsoft.JSInterop.WebAssembly;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Services
+{
+    public class WebAssemblyDynamicResourceLoader
+    {
+        internal const string GetDynamicAssemblies = "window.Blazor._internal.getDynamicAssemblies";
+        internal const string ReadDynamicAssemblies = "window.Blazor._internal.readDynamicAssemblies";
+
+        private static List<string> _loadedAssemblyCache = new List<string>();
+
+        private readonly WebAssemblyJSRuntime _jsRuntime;
+
+        internal WebAssemblyDynamicResourceLoader(WebAssemblyJSRuntime jsRuntime)
+        {
+            _jsRuntime = jsRuntime;
+        }
+
+        public async Task<IEnumerable<Assembly>> LoadDynamicAssemblies(IEnumerable<string> assembliesToLoad)
```

Comment:
```suggestion
        public async Task<IEnumerable<Assembly>> LoadAsync(IEnumerable<string> assembliesToLoad)
```

perhaps?

Reply:
Opted for `LoadAssembliesAsync` instead so the input type is clear.